beancount_import/unbook.py

```diff
@@ -5,20 +5,32 @@
 from beancount.core.number import ZERO, MISSING
 
 def group_postings_by_meta(postings: Iterable[Posting]) -> Iterable[List[Posting]]:
-    """Groups postings that have identical, non-None meta values together.
+    """Groups postings with identical account and non-empty meta values.
 
     This is intended to group together multiple postings generated by the
     booking procedure that correspond to a single original posting.
     """
     prev_meta = None # type: Optional[Meta]
+    prev_acct = ""
     posting_list = []
+    postings = sorted(postings, key=lambda posting: posting.account)
     for posting in postings:
-        if posting.meta is not None and posting.meta is prev_meta:
+        # We need to ignore lineno for comparison purposes, otherwise this
+        # method will never book any postings.
+        post_meta = posting.meta.copy()
+        # Parser meta `filename` and `lineno` metadata should be ignored.
+        if "filename" in post_meta:
+            del post_meta["filename"]
```

Comment:
Do we really want to delete filename? I would think postings split for booking must still always be in the same transaction, therefore same file. Shouldn't matter though, since this function should never be called with postings not from the same transaction.

Reply:
I added it so that postings with no meta on them won't get merged.
Since we are using meta equality instead of identity two postings with no meta on them would otherwise get grouped.